pkg/operator/etcdmemberlistercontroller/etcdmemberlistercontroller.go

```diff
@@ -0,0 +1,86 @@
+package etcdmemberlistercontroller
+
+import (
+	"context"
+	"time"
+
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+	errorsutil "k8s.io/apimachinery/pkg/util/errors"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-etcd-operator/pkg/dnshelpers"
+	"github.com/openshift/cluster-etcd-operator/pkg/etcdcli"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// EtcdMemberListerController keeps etcd members cached, and updates the cache when the membership changes
+type EtcdMemberListerController struct {
+	operatorClient v1helpers.OperatorClient
+	etcdClient     etcdcli.EtcdClient
+	membersCache   map[string]*etcdserverpb.Member
```

Comment:
I think this will need some kind of locking, given this will be consumed by other controllers?